Assets/PatchKit Patcher/Editor/CustomBuilding.cs

```diff
@@ -1,123 +1,154 @@
         {
-            case BuildTarget.StandaloneWindows:
-            case BuildTarget.StandaloneWindows64:
-                return "Patcher.exe";
-
-            case BuildTarget.StandaloneLinux:
-            case BuildTarget.StandaloneLinux64:
-            case BuildTarget.StandaloneLinuxUniversal:
-                return "Patcher";
-
-            case BuildTarget.StandaloneOSXIntel:
-            case BuildTarget.StandaloneOSXIntel64:
-            case BuildTarget.StandaloneOSXUniversal:
-                return "Patcher.app";
-
-            default:
-                return "";
+            Build(BuildTarget.StandaloneOSXIntel);
         }
-    }
 
-    private static void Build(BuildTarget target)
-    {
-        string path = EditorUtility.SaveFolderPanel("Choose where to build the Patcher", "", "");
+        [MenuItem("Tools/Build/OSX x64")]
+        public static void BuildOsx64 ()
+        {
+            Build(BuildTarget.StandaloneOSXIntel64);
+        }
 
-        if (string.IsNullOrEmpty(path))
+        [MenuItem("Tools/Build/OSX Universal")]
+        public static void BuildOsxUniversal ()
         {
-            return;
+            Build(BuildTarget.StandaloneOSXUniversal);
         }
 
-        var patcher = PatchKit.Unity.Patcher.Patcher.Instance;
+        private static string PatcherExecutableName(BuildTarget target)
+        {
+            switch (target)
+            {
+                case BuildTarget.StandaloneWindows:
+                case BuildTarget.StandaloneWindows64:
+                    return "Patcher.exe";
+
+                case BuildTarget.StandaloneLinux:
+                case BuildTarget.StandaloneLinux64:
+                case BuildTarget.StandaloneLinuxUniversal:
+                    return "Patcher";
+
+                case BuildTarget.StandaloneOSXIntel:
+                case BuildTarget.StandaloneOSXIntel64:
+                case BuildTarget.StandaloneOSXUniversal:
+                    return "Patcher.app";
+
+                default:
+                    return "";
+            }
+        }
 
-        if (patcher.EditorAppSecret != PatchKit.Unity.Patcher.Patcher.EditorAllowedSecret)
+        private static void Build(BuildTarget target)
         {
-            if (EditorUtility.DisplayDialog("Error", "Please reset the editor app secret to continue building.", "Reset the secret and continue", "Cancel"))
+            string[] scenePaths = EditorBuildSettings.scenes
+                .Where(s => s.enabled)
+                .Select(s => s.path)
+                .ToArray();
+
+            var scenes = scenePaths
+                .Select(s => EditorSceneManager.OpenScene(s))
+                .ToArray();
+
+            if (scenePaths.Count() == 0)
+            {
+                EditorUtility.DisplayDialog("Error", "Add or enable scenes to be included in the Build Settings menu.", "Ok");
+                return;
+            }
+
+            EditorSceneManager.SaveCurrentModifiedScenesIfUserWantsTo();
```

Comment:
There should be a message why scene should be saved and rejecting saving the scene should stop the building process.

Reply:
This only concerns the currently open scene, which doesn't really need to be saved - as it can be some sort of test or playground scene.

Reply:
That's not good.
If there a scene that is currently open, user need to have at least 3 options:
1. Save scene and continue
2. Do not save scene and continue
3. Cancel the operation

Here there's no way of canceling the operation - you can only save it or lost the changes.

Reply:
![image](https://user-images.githubusercontent.com/5088647/37278451-08d58f04-25e8-11e8-99b4-f3a8e4a4ec07.png)

Tak wygląda dialog po wywołaniu metody `EditorSceneManager.SaveCurrentModifiedScenesIfUserWantsTo()`

Dodam obsługę opcji cancel tej metody.